server/src/main/java/org/opensearch/action/DocWriteRequest.java

```diff
@@ -248,6 +249,22 @@ static DocWriteRequest<?> readDocumentRequest(@Nullable ShardId shardId, StreamI
         return docWriteRequest;
     }
 
+    /**
+     * Validates whether the doc id length is under the limit.
+     * @param id DocId to verify
+     * @param validationException containing all the validation errors.
+     * @return validationException
+     */
+    static ActionRequestValidationException validateDocIdLength(String id, ActionRequestValidationException validationException) {
+        if (id != null && id.getBytes(StandardCharsets.UTF_8).length > 512) {
+            validationException = addValidationError(
```

Comment:
Nitpick, but I hate reassigning method parameter variables. I'd just make this a return statement instead.

Reply:
Agreed, Its an anti-pattern. Let me fix this.